src/Stratis.Bitcoin.Features.PoA/PoAMiner.cs

```diff
@@ -165,10 +165,13 @@ private async Task CreateBlocksAsync()
                 catch (OperationCanceledException)
                 {
                 }
-                // TODO: Find a better way to do this.
-                catch (ConsensusErrorException ce) when (ce.ConsensusError.Code == "invalid-collateral-amount")
+                catch (ConsensusErrorException ce)
                 {
-                    this.logger.LogInformation("Miner failed to mine block due to: '{0}'.", ce.ConsensusError.Message);
+                    // Text from PosMinting:
+                    // All consensus exceptions should be ignored. It means that the miner
+                    // ran into problems while constructing block or verifying it
+                    // but it should not halt the mining operation.
```

Comment:
I don't agree. 

Most of the consensus errors couldn't be spawned unless there is a bug in miner's code. 
There are some consensus errors that depend on other things like data from the counter node, those cases should be handled separately. 

So I don't agree that we should ignore all consensus errors in general.